redux/slices/userSlice/userThunk/finishUserTask.ts

```diff
@@ -21,11 +22,29 @@ export const finishUserTask = createAsyncThunk<
   ThunkApiConfigType
 >(
   "userData/finishUserTask",
-  async ({ taskId, category, taskImportance }, { rejectWithValue }) => {
+  async (
+    { taskId, category, taskImportance },
+    { getState, rejectWithValue }
+  ) => {
     try {
-      const response = await axios.delete(
-        `http://localhost:3000/api/usersTasks?id=${taskId}`
-      );
+      await tasksApi.delete(`?id=${taskId}`);
+      const { userId, doneTasks } = getState().userData;
+
+      await usersApi.put("/login", {
+        userId: userId,
+        doneTasks: {
+          ...doneTasks,
+          categories: {
+            ...doneTasks.categories,
+            [category]: doneTasks.categories[category] + 1,
+          },
+          importanceLevel: {
+            ...doneTasks.importanceLevel,
+            [taskImportance]: doneTasks.importanceLevel[taskImportance] + 1,
+          },
+        },
+      });

```

Comment:
Tutaj można pomyśleć o tym, aby zrobić specjalny endpoint do którego przekażesz po prostu te dwa parametry które sie zmieniają, zamiast pakować wszystko nawet to, co zostaje niezmienne. Nie wiem skąd masz API jbc